andela/ah-alpha
# #162948915 Users can comment on articles

#### What does this PR do?
Has the functionality for the user to comment on articles

#### Description of Task(s) to be completed?
* Users should be able to comment on an article, view comments for an article, edit their comment on an article and delete an article

#### How should this be manually tested?
* Pull the branch `git fetch origin ft-comments-162948915:ft-comments-162948915`
* Checkout to the branch `git checkout ft-comments-162948915`
* Install requirements `pip3 install -r requirements.txt`
* Run the server `python3 manage.py runserver`
* Launch postman and create a user by sending a POST request to `http://127.0.0.1:8000/api/v1/users/` with a body of:
```{
   "user":{
        "username": "username",
        "email": "example@email.com",
        "password": "Passw0rd123"
    }
}```
* Login by sending a POST request to `http://127.0.0.1:8000/api/v1/users/login/` with your registered email and password
```{
   "user":{
        "email": "example@email.com",
        "password": "Passw0rd123"
    }
}```
* Copy the returned JWT token and store it somewhere
* Use postman to create an article by sending a POST request to `http://127.0.0.1:8000/api/v1/articles`
* The body should have the format of
```{ 
  "article":{
       "image_path":"....",
            "title":"mytitle",
            "description":"and",
            "body":"your content"
    }
}
```
* Select Bearer Token under Authorization and paste in the token gotten previously.
* The response should be similar to
```{
    "id": 14,
    "author": {
        "email": "example@gmail.com",
        "username": "user"
    },
    "image_path": ".....",
    "slug": "rick-5",
    "title": "mytitle",
    "description": "and",
    "body": "your content",
    "created_at": "2019-01-23T14:51:16.512539Z",
    "updated_at": "2019-01-23T14:51:16.512733Z",
    "favourites": false
}
```
* Send a post request to **add comment** `http://127.0.0.1:8000/api/v1/articles/<slug>/comments`. Replace the <slug> param with the value of the slug from the out returned on creating an article.
* To **edit** a comment, send a put request with `http://127.0.0.1:8000/api/v1/<slug>/comments/id/, replacing <slug> with the actual slug and ID with the comment ID to update.
* To **delete** a comment, send a delete request with `http://127.0.0.1:8000/api/v1/<slug>/comments/id/, replacing <slug> with the actual slug and ID with the comment ID to update.

#### What are the relevant pivotal tracker stories?
[#162948915](https://www.pivotaltracker.com/story/show/162948915)
